packages/cli/src/util/output/create-output.ts

```diff
@@ -25,31 +26,62 @@ export interface LinkOptions {
   fallback?: false | (() => string);
 }
 
+let defaultChalkColorLevel: chalk.Level = 0;
+
 export class Output {
-  stream: tty.WriteStream;
-  debugEnabled: boolean;
-  supportsHyperlink: boolean;
-  colorDisabled: boolean;
+  stream!: tty.WriteStream;
+  debugEnabled!: boolean;
+  supportsHyperlink!: boolean;
+  colorDisabled!: boolean;
   private spinnerMessage: string;
   private _spinner: StopSpinner | null;
 
-  constructor(
-    stream: tty.WriteStream,
-    {
-      debug: debugEnabled = false,
-      supportsHyperlink = detectSupportsHyperlink(stream),
-      noColor = false,
-    }: OutputOptions = {}
-  ) {
-    this.stream = stream;
-    this.debugEnabled = debugEnabled;
-    this.supportsHyperlink = supportsHyperlink;
+  constructor(stream: tty.WriteStream, options: OutputOptions = {}) {
     this.spinnerMessage = '';
     this._spinner = null;
 
-    this.colorDisabled = getNoColor(noColor);
-    if (this.colorDisabled) {
-      chalk.level = 0;
+    this.initialize({
+      ...options,
+      stream,
+    });
+  }
+
+  /**
+   * Parts of the constructor logic that can be called again after construction
+   * to change some values.
+   */
+  initialize({
+    stream,
+    debug: debugEnabled,
+    supportsHyperlink,
+    noColor,
+  }: OutputOptions = {}) {
+    if (stream !== undefined) {
+      this.stream = stream;
+    }
+
+    if (debugEnabled !== undefined) {
+      this.debugEnabled = debugEnabled;
+    }
+
+    if (supportsHyperlink !== undefined) {
+      this.supportsHyperlink = supportsHyperlink;
+    }
+
```

Comment:
Non-blocking: This case is already handled below.

```suggestion
```

Reply:
Ah, copy paste error. Thanks!